assets/queries/k8s/hpa_deployments_no_replicas/query.rego

```diff
@@ -0,0 +1,38 @@
+package Cx
+
+CxPolicy [ result ] {
+  
+  metadata := input.document[i].metadata
+  spec := input.document[i].spec
+  kindCheck := input.document[i].kind
+  replicasCheck := spec.replicas
+    
+  replicasCheck > 0 
+  kindCheck == "Deployment"
```

Comment:
You also need to check if this Deployment is targeted by HPA.
For example, this case is positive:
```
apiVersion: apps/v1
kind: Deployment
metadata:
  name: php-apache-1
spec:
  replicas: 1
  ...
---
apiVersion: autoscaling/v2beta2
kind: HorizontalPodAutoscaler
metadata:
  name: php-apache
spec:
  scaleTargetRef:
    apiVersion: apps/v1
    kind: Deployment
    name: php-apache
...
```
There must be a HorizontalPodAutoscaler that targets the Deployment. Which means you have to check if scaleTargetRef.kind equals "Deployment" and scaleTargetRef.name equals the name of the Deployment you are checking.

Then you just need to check if Deployment.spec.replicas is defined. You don't need to check if it is bigger than 0.